src/transaction/lock_table.c

```diff
@@ -117,75 +123,81 @@ LOCK_COMPATIBILITY lock_Comp[11][11] = {
  *
- *   SCH-S | N/A     N/A  SCH-S  SCH-S     IS      S     IX    SIX    N/A      X  SCH-M
+ *      IS | N/A     N/A     IS     IS     IS      S     IX    SIX    N/A      X  SCH-M        IS
  *
- *      IS | N/A     N/A     IS     IS     IS      S     IX    SIX    N/A      X  SCH-M
+ *       S | N/A     N/A      S      S      S      S    SIX    SIX      U      X  SCH-M         S
  *
- *       S | N/A     N/A      S      S      S      S    SIX    SIX      U      X  SCH-M
+ *      IX | N/A     N/A     IX     IX     IX    SIX     IX    SIX    N/A      X  SCH-M        IX
  *
- *      IX | N/A     N/A     IX     IX     IX    SIX     IX    SIX    N/A      X  SCH-M
+ *     SIX | N/A     N/A    SIX    SIX    SIX    SIX    SIX    SIX    N/A      X  SCH-M       SIX
  *
- *     SIX | N/A     N/A    SIX    SIX    SIX    SIX    SIX    SIX    N/A      X  SCH-M
+ *       U | N/A     N/A      U    N/A    N/A      U    N/A    N/A      U      X    N/A       N/A
  *
- *       U | N/A     N/A      U    N/A    N/A      U    N/A    N/A      U      X    N/A
+ *       X | N/A     N/A      X      X      X      X      X      X      X      X  SCH-M         X
  *
- *       X | N/A     N/A      X      X      X      X      X      X      X      X  SCH-M
+ *   SCH-M | N/A     N/A  SCH-M  SCH-M  SCH-M  SCH-M  SCH-M  SCH-M    N/A  SCH-M  SCH-M     SCH-M
  *
- *   SCH-M | N/A     N/A  SCH-M  SCH-M  SCH-M  SCH-M  SCH-M  SCH-M    N/A  SCH-M  SCH-M
+ * BU_LOCK | N/A     N/A     BU      X     IS      S     IX    SIX    N/A      X  SCH-M        BU
  * ------------------------------------------------------------------------------------------------
  * N/A : not applicable
  */
 
-LOCK lock_Conv[11][11] = {
+LOCK lock_Conv[12][12] = {
   /* N/A */
   {NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK,
-   NA_LOCK, NA_LOCK, NA_LOCK}
+   NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK}
   ,
   /* NON2PL */
   {NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK,
-   NA_LOCK, NA_LOCK, NA_LOCK}
+   NA_LOCK, NA_LOCK, NA_LOCK, NA_LOCK}
   ,
   /* NULL */
   {NA_LOCK, NA_LOCK, NULL_LOCK, SCH_S_LOCK, IS_LOCK, S_LOCK, IX_LOCK,
-   SIX_LOCK, U_LOCK, X_LOCK, SCH_M_LOCK}
+   SIX_LOCK, U_LOCK, X_LOCK, SCH_M_LOCK, BU_LOCK}
   ,
   /* SCH-S */
   {NA_LOCK, NA_LOCK, SCH_S_LOCK, SCH_S_LOCK, IS_LOCK, S_LOCK, IX_LOCK,
-   SIX_LOCK, NA_LOCK, X_LOCK, SCH_M_LOCK}
+   SIX_LOCK, NA_LOCK, X_LOCK, SCH_M_LOCK, X_LOCK}
   ,
   /* IS */
   {NA_LOCK, NA_LOCK, IS_LOCK, IS_LOCK, IS_LOCK, S_LOCK, IX_LOCK, SIX_LOCK,
-   NA_LOCK, X_LOCK, SCH_M_LOCK}
+   NA_LOCK, X_LOCK, SCH_M_LOCK, IS_LOCK}
   ,
   /* S */
   {NA_LOCK, NA_LOCK, S_LOCK, S_LOCK, S_LOCK, S_LOCK, SIX_LOCK, SIX_LOCK,
-   U_LOCK, X_LOCK, SCH_M_LOCK}
+   U_LOCK, X_LOCK, SCH_M_LOCK, IS_LOCK}
   ,
   /* IX */
   {NA_LOCK, NA_LOCK, IX_LOCK, IX_LOCK, IX_LOCK, SIX_LOCK, IX_LOCK, SIX_LOCK,
-   NA_LOCK, X_LOCK, SCH_M_LOCK}
+   NA_LOCK, X_LOCK, SCH_M_LOCK, IS_LOCK}
```

Comment:
BU + IX = BU
IX + BU = BU

Reply:
> IX + BU = BU

If `IS + BU = X`, is it alright to have `IX + BU = BU`? Since IX > IS, it is counter intuitive to have IS + BU < IX + BU.

Reply:
That's also true indeed.